app/engines/search_engine_adapter.rb

```diff
@@ -24,18 +26,11 @@ def initialize(klass, options)
   def_instance_delegator :@search_engine_response, :end_record, :endrecord
 
   def run
-    instrument_name = "#{@search_engine.class.name.tableize.singularize}.usasearch"
-    ActiveSupport::Notifications.instrument(instrument_name,
-                                            query: { term: @search_engine.query }) do
-      @search_engine_response = @search_engine.execute_query
-    end
-  rescue SearchEngine::SearchError => error
-    Rails.logger.warn "Error getting image search results from #{@search_engine.class} endpoint: #{error}"
-    false
+    @search_engine_response = search
   end
 
   def results
-    @results || (paginate(post_process_results(@search_engine_response.results)) if @search_engine_response.results)
+    @results || (paginate(post_process_results(@search_engine_response.results)) if @search_engine_response && @search_engine_response.results)
```

Comment:
I think you can clean this up with the "safe operator":
`if @search_engine_response&.results`

Reply:
Fixed!

Reply:
I'm curious - what necessitated the reversion?

Reply:
In the case of an error, `SearchEngineAdapter.run` returns false making `@search_engine_response` false, and false doesn't have a `.results` method.

Reply:
Ah, gotcha. Thanks, that makes sense.